### PR TITLE
[BottomNavigation] Adds the Ripple behavior to BottomNavigation. 

### LINKED ISSUE #7377: [Ripple] Internal issue: b/132669181
This was filed as an internal issue. If you are a Googler, please visit [b/132669181](http://b/132669181) for more details.

---
#### Internal data
- Associated internal bug: [b/132669181](http://b/132669181)

### PR DESCRIPTION
**Pull request by yarneo**
## Related links
* Bug: Closes #7377  
* Ripple: [MDCRippleTouchController](https://github.com/material-components/material-components-ios/tree/develop/components/Ripple)

## Introduction
This PR integrates [`MDCRippleTouchController`](https://github.com/material-components/material-components-ios/blob/develop/components/Ripple/src/MDCRippleTouchController.h) into [`MDCBottomNavigationBar`](https://github.com/material-components/material-components-ios/tree/develop/components/BottomNavigation). This behavior is an opt in flag property added to MDCBottomNavigationBar, `enableRippleBehavior`.  This improves the visual touch feedback for our users as MDCInkView didn't support our current motion guidelines.

## Videos
| Before | After |
| - | - |
|oldRipple|newRipple|

### CODE REVIEW

File: components/BottomNavigation/src/MDCBottomNavigationBar.m [hunk 1]
```diff
@@ -480,13 +481,6 @@ - (UITabBarItem *)tabBarItemForPoint:(CGPoint)point {
 
 #pragma mark - Touch handlers
 
-- (void)didTouchDownButton:(UIButton *)button {
-  MDCBottomNavigationItemView *itemView = (MDCBottomNavigationItemView *)button.superview;
-  CGPoint centerPoint =
-      CGPointMake(CGRectGetMidX(itemView.inkView.bounds), CGRectGetMidY(itemView.inkView.bounds));
-  [itemView.inkView startTouchBeganAnimationAtPoint:centerPoint completion:nil];
-}
-
```
Comment (jverkoey) [inline components/BottomNavigation/src/MDCBottomNavigationBar.m]: Can we move these deletions to a separate PR?

Comment (yarneo) [reply components/BottomNavigation/src/MDCBottomNavigationBar.m]: yes! will do

**Review by jverkoey — APPROVED** [review]
